### PR TITLE
GDAL 3.7.0

### PR DESCRIPTION
**Pull request by akrherz**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
*  Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

@conda-forge-admin please rerender

I'll create the 3.6 branch prior to merging this, or somebody else can beat me to it :)

### CODE REVIEW

**Comment by rouault** [comment]
build failure on Python bindings on osx_arm64:
```
+ cmake '-UPython*' -DPython_LOOKUP_VERSION=3.11.3 -DBUILD_PYTHON_BINDINGS:BOOL=ON -DGDAL_USE_EXTERNAL_LIBS=OFF -DGDAL_BUILD_OPTIONAL_DRIVERS:BOOL=OFF -DOGR_BUILD_OPTIONAL_DRIVERS:BOOL=OFF /Users/runner/miniforge3/conda-bld/gdal-split_1683748194507/work
-- Command "/Users/runner/miniforge3/conda-bld/gdal-split_1683748194507/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehol/bin/swig -swiglib" failed with output:

-- Could NOT find SWIG (missing: SWIG_DIR) 
```

The check for SWIG must be done in CMake's FindSWIG.cmake : https://github.com/Kitware/CMake/blob/master/Modules/FindSWIG.cmake#L94

Wondering if the issue wouldn't be that the swig binary used here is a arm64 one, instead of being the one of the build platform ?

**Comment by akrherz** [comment]
With my manual SWIG_DIR setting, we now get this error
```
2023-05-11T03:39:03.8227030Z ++ swig -swiglib
2023-05-11T03:39:03.8245000Z /Users/runner/miniforge3/conda-bld/gdal-split_1683773142397/work/install_python.sh: line 16: /Users/runner/miniforge3/conda-bld/gdal-split_1683773142397/...bin/swig: Bad CPU type in executable
```

**Comment by rouault** [comment]
> With my manual SWIG_DIR setting, we now get this error

confirms the architecture error.

Wouldn't moving the "swig" item from the "host" section of https://github.com/conda-forge/gdal-feedstock/blob/main/recipe/meta.yaml#L267 to "swig # [build_platform != target_platform]" in the "build" section close to https://github.com/conda-forge/gdal-feedstock/blob/main/recipe/meta.yaml#L254 fix the issue ?

File: recipe/meta.yaml [hunk 1]
```diff
@@ -264,7 +268,6 @@ outputs:
         # Workaround to prevent conflicts seen in
         #  https://github.com/conda-forge/gdal-feedstock/pull/543
         - openssl
-        - swig
```
Comment (rouault) [inline recipe/meta.yaml:267]: Actually from the build error on https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=703871&view=results, it looks like swig must perhaps be kept in the host section, in addition to being in the build one ?

**Comment by akrherz** [comment]
@conda-forge-admin please rerender

File: recipe/meta.yaml (hunk 1 above)
Comment (akrherz) [reply recipe/meta.yaml]: trying

**Comment by akrherz** [comment]
I am unsure what is happening here.  As I review previous successful GDAL builds, I do find the same swig errors reports, but the compilation works somehow.  Am continuing to boggle.

**Comment by rouault** [comment]
> As I review previous successful GDAL builds, I do find the same swig errors reports, but the compilation works somehow

yes that's expected. GDAL tarballs of previous versions bundled SWIG generated files. Starting with 3.7.0, SWIG must be available at build time to generate those SWIG generated files.

**Comment by jschueller** [comment]
swig dependency must be moved from host to build section

**Comment by akrherz** [comment]
> swig dependency must probably be moved from host to build section

Thank you, trying.

**Comment by akrherz** [comment]
At least on Linux64, I downloaded the `swig-4.1.1-h799ce0a` package and inspected the swig binary like so:
```
$ file bin/swig 
bin/swig: Mach-O 64-bit arm64 executable, flags:<NOUNDEFS|DYLDLINK|TWOLEVEL|WEAK_DEFINES|BINDS_TO_WEAK|PIE>
```
That looks good, but looking at the failing build logs, we appear to be getting `swig-4.1.1-h071a223`, which is the wrong arch.  Hopefully @jschueller got us on the right track here, thanks again!  One of these days, I fully grok the build/host differences when doing cross-compiles...

**Comment by akrherz** [comment]
> does it ?

Sorry again, I horribly worded my message.  I downloaded the osx arm65 artifact and am denoting what `file` reported about the binary on linux64 as I have no mac to play with.

**Comment by akrherz** [comment]
Holy cow, this may turn green :)  Anybody else on @conda-forge/gdal wish to merge this PR?

**Pull request merged**